app/scripts/modules/netflix/pipeline/stage/isolatedTestingTarget/isolatedTestingTargetStage.ts

```diff
@@ -0,0 +1,252 @@
+import {module} from 'angular';
+import {filter, flatten, map, each} from 'lodash';
+import {IModalService} from 'angular-ui-bootstrap';
+
+import {AuthenticationService} from 'core/authentication/authentication.service';
+import {CLOUD_PROVIDER_REGISTRY, CloudProviderRegistry} from 'core/cloudProvider/cloudProvider.registry';
+import {ICluster, IStage, ServerGroup} from 'core/domain/index';
+import {SERVER_GROUP_COMMAND_BUILDER_SERVICE} from 'core/serverGroup/configure/common/serverGroupCommandBuilder.service';
+import {SERVER_GROUP_READER, ServerGroupReader} from 'core/serverGroup/serverGroupReader.service';
+import {NAMING_SERVICE, NamingService} from 'core/naming/naming.service';
+import {EDIT_VIP_MODAL_CONTROLLER} from './editVip.modal.controller';
+
+interface IVipOverride {
+  oldVip: string;
+  oldSecureVip: string;
+  newVip: string;
+  newSecureVip: string;
+  [key: string]: string; // This makes the linter happy when an object of shape IVipOverride is referenced with a variable instead of dot notation
+}
+
+interface IIsolatedTestingTargetStage extends IStage {
+  clusters: ICluster[];
+  vipOverrides: {
+    [clusterId: string]: IVipOverride;
+  };
+  username: string;
+}
+
+class IsolatedTestingTargetStageCtrl {
+  static get $inject() {
+    return ['$scope', '$uibModal', 'stage', 'namingService', 'providerSelectionService',
+            'authenticationService', 'cloudProviderRegistry', 'serverGroupCommandBuilder',
+            'serverGroupReader', 'awsServerGroupTransformer'];
+  }
+
+  constructor(private $scope: any,
+              private $uibModal: IModalService,
+              private stage: IIsolatedTestingTargetStage,
+              private namingService: NamingService,
+              private providerSelectionService: any,
+              public authenticationService: AuthenticationService,
```

Comment:
i don't see this used outside the ctor, i think we can remove the access modifier completely.